### ISSUE TITLE
Panic in certigo/lib.explainCipher

### ISSUE THREAD

**Issue by csstaub** [square/certigo#202]
```
❯❯❯ certigo connect <redacted>
panic: runtime error: index out of range [1] with length 1

goroutine 1 [running]:
github.com/square/certigo/lib.explainCipher(0xc00002a960, 0xc, 0xc00002a960, 0xc, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/private/tmp/certigo-20200106-25422-1jnrsi9/certigo-1.11.0/lib/tls.go:182 +0x1e3
github.com/square/certigo/lib.EncodeTLSInfoToText(0xc0001a80b0, 0x0, 0x5, 0x0)
	/private/tmp/certigo-20200106-25422-1jnrsi9/certigo-1.11.0/lib/tls.go:62 +0x148
main.main()
	/private/tmp/certigo-20200106-25422-1jnrsi9/certigo-1.11.0/main.go:162 +0x667
```

This happened on a new AWS Classic LB, haven't been able to consistently reproduce.

**Comment by mbyczkowski** [comment]
@csstaub are you running 1.11 or have you built from master? There hasn't been a new release since Nov '18.

**Comment by jason-riddle** [comment]
I ran into this issue as well and did some debugging and there are two problems.

1. On v1.11.0, `lib.lookup` returns `UNKNOWN_304` for TLS 1.3. `lib.explainCipher` does a
   split on `_WITH_` but the array is only populated with one element and it
   is assumed that `kexAndCipher` has two elements.
   > Note: This was fixed in master @ https://github.com/square/certigo/commit/79de7d2a354a6cc5ac1457dc955e592c702f6afd
3. `lib.description` is missing support for some TLS 1.3 cipher suites.

## Quick, insecure fix

Checkout `master` branch.

### Patch

`tls13_ciphers.patch`

```diff
diff --git a/lib/tls.go b/lib/tls.go
index a9fefcd..5c0855d 100644
--- a/lib/tls.go
+++ b/lib/tls.go
@@ -206,4 +206,10 @@ var cipherSuites = map[uint16]description{
 	tls.TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384: {"", "TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384", good},
 	tls.TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305:    {"", "TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305", good},
 	tls.TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305:  {"", "TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305", good},
+
+	tls.TLS_AES_128_GCM_SHA256:       {"", "TLS_AES_128_GCM_SHA256", good},
+	tls.TLS_AES_256_GCM_SHA384:       {"", "TLS_AES_256_GCM_SHA384", good},
+	tls.TLS_CHACHA20_POLY1305_SHA256: {"", "TLS_CHACHA20_POLY1305_SHA256", good},
+
+	tls.TLS_FALLBACK_SCSV: {"", "TLS_FALLBACK_SCSV", insecure},
 }

```

### Apply

```
git apply tls13_ciphers.patch
```

### Build

```
go build -o certigo-with-tls13-ciphers
```

### Validate

> TODO: Why is Cipher Suite still empty? Maybe this isn't fixed yet..

```
./certigo-with-tls13-ciphers connect --verbose squareup.com:443
** TLS Connection **
Version: TLS 1.3
Cipher Suite:

...
```

## Possible Future TODO

As of go1.14rc1 (https://github.com/golang/go/commit/0ee22d97ad1d1d40df1a58b0efae022e4f82f1b5), it looks like there now exists a `tls.CipherSuites` which looks something like

```
func CipherSuites() []*CipherSuite {
	return []*CipherSuite{
		{TLS_RSA_WITH_3DES_EDE_CBC_SHA, "TLS_RSA_WITH_3DES_EDE_CBC_SHA", supportedUpToTLS12, false},
		{TLS_RSA_WITH_AES_128_CBC_SHA, "TLS_RSA_WITH_AES_128_CBC_SHA", supportedUpToTLS12, false},
		...
		{TLS_AES_128_GCM_SHA256, "TLS_AES_128_GCM_SHA256", supportedOnlyTLS13, false},
		{TLS_AES_256_GCM_SHA384, "TLS_AES_256_GCM_SHA384", supportedOnlyTLS13, false},
		...
	}
}
```

It may prove useful to use this instead of maintaining a map of hand-maintained cipher suites when go1.14 is released.

**Comment by mbyczkowski** [comment]
@jason-riddle thanks for looking into this! 💯 Would you like to turn your patch into a PR?

**Comment by jason-riddle** [comment]
Sure maybe sometime this week or next.

**Comment by mbyczkowski** [comment]
This has been fixed via #208 and released in 1.12.1
